### ISSUE TITLE
[WCS 2.5] Maybe update the unscheduling function to use Action Scheduler

### ISSUE THREAD

**Issue by james-allan** [Prospress/woocommerce-subscriptions-disable-report-cache-updates#1]
As part of https://github.com/Prospress/woocommerce-subscriptions/pull/2980 which will be released in 2.5.0, we were looking at changing the scheduling of report cache updates via Cron and `wp_schedule_single_event()`  to use the action scheduler instead. This means this plugin will need to update the function it calls to unschedule the cache update event or more reliably use the `wcs_report_cache_updates_enabled` filter.

This might be a really big issue as I suspect there are a lot of customers using this plugin and so we might have to scrap that idea. 

I'm just making a note with this PR to make sure we don't forget to do that.

**Comment by thenbrent** [comment]
> This might be a really big issue as I suspect there are a lot of customers using this plugin and so we might have to scrap that idea.

Good catch. To mitigate that, we might want to look at releasing that plugin via WordPress.org so that we can provide an easy upgrade / new version notification to folks.

I'm pretty sure if we're able to use the same file name as the plugin's slug on WordPress.org, each WordPress installation where the plugin is installed from GitHub will automatically detect an update.

The gotchas with that are:

1. the plugin folder likely includes `-master` suffix, as GitHub appends that
1. trademark issues might prevent us using a slug prefixed with `woocommerce-`

**Comment by thenbrent** [comment]
Fixed with #4.
